js/admin/user-form-settings.js

```diff
@@ -0,0 +1,190 @@
+      if (isChecked) {
+        $(row).addClass('delete-in-progress');
+      } else {
+        $(row).removeClass('delete-in-progress');
+      }
+    });
+    $('.update-field').on('input', function (value) {
+      let row = value.target.closest("div[name='field-row']");
+      let resetButton = $(row).find("input[name='reset-changes-button']")[0];
+
+      $(row).addClass('update-in-progress');
+      $(resetButton).attr('disabled', false);
+
+      let fields = $(row).find("input[type='text']");
+      let isNoChanges = true;
+      let i = 0;
+
+      while (isNoChanges && i < fields.length) {
+        [prefix, property, field_name, lang] = fields[i].name.split('/');
+        isNoChanges = fields[i].value === user_form_fields[field_name][property][lang];
+        i++;
+      }
+      if (isNoChanges) {
+        $(resetButton).attr('disabled', true);
+        $(row).removeClass('update-in-progress');
+      }
+    });
+    $('input[name="reset-changes-button"]').on('click', function(value) {
+      let row = value.target.closest("div[name='field-row']");
+      $(row).removeClass('update-in-progress');
+      $(value.target).attr('disabled', true);
+
+      let fields = $(row).find("input[type='text']");
+      for (let i = 0; i < fields.length; i++) {
+        [prefix, property, field_name, lang] = fields[i].name.split('/');
+        fields[i].value = user_form_fields[field_name][property][lang] || '';
+      }
+    });
+    $('#btns-field-type-choice button').on('click', function(value) {
+      $('button').removeClass('selected');
+      $(this).addClass('selected');
+
+      $('.new_field_row').css('display', 'table-row');
+      $('label[name="name"]').css('display', 'block');
+      $('label[name="instruction"]').css('display', 'block');
+
+      $('label[name="options"]').css('display', 'none');
+      $('label[name="options"]').children('.required_input').removeAttr("required");
+      $('label[name="regex"]').css('display', 'none');
+      $('label[name="unit"]').css('display', 'none');
+
+      $('#multiple').css('display', 'none');
+
+      let type = value.target.name;
+      document.getElementById("new-field-type").value = type;
+      switch (type) {
+        case 'text':
+        case 'email':
+        case 'tel':
+        case 'checkbox':
+          break;
+        case 'number':
+          $('label[name="unit"]').css('display', 'block');
+          break;
+        case 'radio':
+          $('label[name="options"]').css('display', 'block');
+          $('label[name="options"]').children('.required_input').attr("required", true);
+          break;
+        case 'select':
+          $('label[name="options"]').css('display', 'block');
+          $('#multiple').css('display', 'table-row');
+          $('label[name="options"]').children('.required_input').attr("required", true);
+          break;
+      }
+    });
+    $('#copy-shortcode').on('click', function() {
+      let copyText = document.getElementById("form-shortcode");
```

Comment:
$(this).first() ou $(this).next() ?

De manière générale il vaut mieux essayer de multiplier les appels DOM trop couteux